xbmc/addons/gui/skin/SkinTimers.cpp

```diff
@@ -0,0 +1,303 @@
+/*
+ *  Copyright (C) 2022 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+#include "SkinTimers.h"
+
+#include "GUIInfoManager.h"
+#include "ServiceBroker.h"
+#include "guilib/GUIComponent.h"
+#include "interfaces/builtins/Builtins.h"
+#include "utils/XBMCTinyXML.h"
+#include "utils/log.h"
+
+#include <chrono>
+
+using namespace std::chrono_literals;
+
+CSkinTimer::CSkinTimer(const std::string& name,
+                       const INFO::InfoPtr startCondition,
+                       const INFO::InfoPtr stopCondition,
+                       const std::string& startAction,
+                       const std::string& stopAction,
+                       bool resetOnStart)
+{
+  m_name = name;
+  m_startCondition = startCondition;
+  m_stopCondition = stopCondition;
+  m_startAction = startAction;
+  m_stopAction = stopAction;
+  m_reset = resetOnStart;
+}
+
+void CSkinTimer::Start()
+{
+  if (m_reset)
+  {
+    CStopWatch::StartZero();
+  }
+  else
+  {
+    CStopWatch::Start();
+  }
+  OnStart();
+}
+
+void CSkinTimer::Stop()
+{
+  CStopWatch::Stop();
+  OnStop();
+}
+
+bool CSkinTimer::VerifyStartCondition()
+{
+  return m_startCondition && m_startCondition->Get();
+}
+
+bool CSkinTimer::VerifyStopCondition()
+{
+  return m_stopCondition && m_stopCondition->Get();
+}
+
+INFO::InfoPtr CSkinTimer::GetStartCondition()
+{
+  return m_startCondition;
+}
+
+INFO::InfoPtr CSkinTimer::GetStopCondition()
+{
+  return m_stopCondition;
+}
+
+void CSkinTimer::OnStart()
+{
+  if (!m_startAction.empty())
+  {
+    CBuiltins::GetInstance().Execute(m_startAction);
+  }
+}
+
+void CSkinTimer::OnStop()
+{
+  if (!m_stopAction.empty())
+  {
+    CBuiltins::GetInstance().Execute(m_stopAction);
+  }
+}
+
+CSkinTimerManager::CSkinTimerManager() : CThread("SkinTimers")
+{
+}
+
+void CSkinTimerManager::Start()
+{
+  if (m_timers.empty())
+  {
+    return;
+  }
+
+  Create();
+}
+
+void CSkinTimerManager::LoadTimers(const std::string& path)
+{
+  CXBMCTinyXML doc;
+  if (!doc.LoadFile(path))
+  {
+    CLog::LogF(LOGINFO, "Error loading timers file {}: {} (row: {}, col: {})", path,
```

Comment:
LOGERROR, not INFO

Reply:
`LOGWARNING` probably better here as the file may not exist (and we removed the exist check)